third_party/jni/emjvm.js

```diff
@@ -58,8 +58,9 @@ mergeInto(LibraryManager.library, {
     var obj = EmJVM.objects[string];
     assert(obj.name == 'string');
     if (isCopy) setValue(isCopy, 'i8', 1);
-    var buffer = _malloc(obj.value.length+1);
```

Comment:
In locations like these, the old code pattern was wrong if the string happened to contain non-ASCII UTF-8 characters, since this assumed that each character would take up exactly one byte. Instead, the `lengthBytesUTF8` function needs to be used to correctly count the byte length when encoded to UTF-8.